[Doc] Correctly cite setup-passwords CLI for minimal security guide

ywangd opened the pull request:
The minimal security guide is meant to use setup-passwords instead of reset-password. This is consistent with the 7.x version.

In longer term, we might actually want to replace setup-passwords with reset-password. But it requires rewriting of the associated the guide while this PR fixes the immediate problem of incorrect usage of CLI command.

Relates: #88393

ywangd committed:
[Doc] Correctly cite setup-passwords CLI for minimal security guide

The minimal security guide is meant to use setup-passwords instead of
reset-password. This is consistent with the 7.x version.

In longer term, we might actually want to replace setup-passwords with
reset-password. But it requires rewriting of the associated the guide
while this PR fixes the immediately problem of incorrect usage of CLI
command.

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

elasticsearchmachine commented:
Pinging @elastic/es-security (Team:Security)

lockewritesdocs commented:
>The minimal security guide is meant to use setup-passwords instead of reset-password. 

@ywangd, I'm curious why we want to recommend `elasticsearch-setup-passwords` in this particular page when we use `elasticsearch-reset-password` in every other context relating to resetting passwords. Is this a technical limitation? I'm just trying to understand why this mention is incorrect.

ywangd commented:
@lockewritesdocs I agree `elasticsearch-reset-password` should be prefered in 8.x. I am reverting it back to `setup-passwords` on this page because descriptions on the page are for `setup-passwords`, e.g.:
* `However, you should only run this utility *one time* for the entire cluster.` There limitation is for setup-passwords only
* `/bin/elasticsearch-reset-password auto` (and the `interactive` variant). This is incorrect for reset-password CLI because it requires a username to be passed in with `-u`, e.g. `-u kibana_system`.

Also `setup-passwords` configures passwords for all builtin users including `kibana_system`, `beats_system` etc. But `reset-password` works on one user at a time. For this particular page, configuring `kibana_system` is sufficient. But in the "Set up basic security plus HTTPS", we also [suggest](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-basic-setup-https.html#beats-monitoring-role) to use the `beats_system` user. So maybe we should reset-password for all builtin users like setup-passwords does.

Instead of addressing all above issues, I took the shortcut to just revert the command to setup-passwords as commented in the PR description:

> In longer term, we might actually want to replace setup-passwords with reset-password. But it requires rewriting of the associated the guide while this PR fixes the immediate problem of incorrect usage of CLI command.

This could be the wrong tradeoff. I am happy for you to discard this PR and rewrite this section to suit `reset-password`. 

PS: Context is https://discuss.elastic.co/t/manually-configure-security-divergence-in-the-documentation/315601

Commit:
Update minimal security and security + HTTPS to use reset-password, plus update links

lockewritesdocs commented:
@ywangd, thanks for the explanation. I think it's best to use `elasticsearch-reset-password` across our documentation, especially since it's also the command that we indicate in the [Elasticsearch startup message](https://github.com/elastic/elasticsearch/blob/be7c7415627377a1b795400fb8dfcc6cbdf0e322/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/InitialNodeSecurityAutoConfiguration.java#L263). I updated the page for minimal security, and also updated the page for security + HTTPS for resetting the `beats_system` user password, plus updated some old links and made some minor changes.

lockewritesdocs commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'main' into doc-fix-reset-vs-setup

ywangd commented on x-pack/docs/en/security/securing-communications/security-minimal-setup.asciidoc:
```diff
@@ -51,14 +51,15 @@ discovery.type: single-node
-username and password are rejected.
+To communicate with your cluster, you must configure a password for
+the `elastic` and `kibana_system` built-in users. Unless you enable anonymous
+access (not recommended), all requests that don’t include a username and
+password are rejected.
```
I think `username and password` is too specific. Might want to replace it with "credentials" since we support more than just username password.

ywangd commented:
@lockewritesdocs Thanks for working on this PR and make it a much better improvement. How should we do the review work? The changes look good to me. But I am still the "original author" ... So probably we just follow the existing procedure and have you approve the changes (on behalf of both of us)?

Commit:
Change 'username and password' to 'credentials'

lockewritesdocs reviewed (approved):
LGTM with the latest changes. Thanks @ywangd!

lockewritesdocs commented:
> But I am still the "original author" ... So probably we just follow the existing procedure and have you approve the changes (on behalf of both of us)?

Works for me! I approved the changes, so either you can merge them or I can -- let me know which you prefer.

ywangd commented:
@elasticmachine run elasticsearch-ci/docs

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [8.5](https://github.com/elastic/elasticsearch/pull/90779)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/90779">](https://github.com/elastic/elasticsearch/pull/90779) |
| ✅ |  [8.4](https://github.com/elastic/elasticsearch/pull/90780)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/90780">](https://github.com/elastic/elasticsearch/pull/90780) |
